In [65]:
import pyspark

try: 
    type(sc)
except NameError:
    sc = pyspark.SparkContext('local[*]')    

datos = sc.textFile('events.csv',8) #8 particiones


In [66]:
a= datos.take(2)
a[1]

'2018-05-31 23:38:05,ad campaign hit,0004b0a2,/comprar/iphone/iphone-5s,,,,,,,,,criteo,,,,,,,,,,'

In [67]:
datos.count()

1011289

# PUNTO 1

In [68]:
a = datos.map(lambda a:a.split(",")).map(lambda a: (a[1],1)).reduceByKey(lambda a,b: a+b).takeOrdered(1,key = lambda a: -a[1])

In [69]:
a

[('viewed product', 528931)]

el evento que mas aparece es viewed product con 528931 apariciones

# PUNTO 2

In [70]:
b = datos.map(lambda a:a.split(",")).filter(lambda a: a[1]=="viewed product").map(lambda a: (a[5],1)).reduceByKey(lambda a,b: a+b).takeOrdered(5,key = lambda a: -a[1])
b

[('iPhone 6', 47538),
 ('iPhone 5s', 42045),
 ('iPhone 6S', 36930),
 ('iPhone 7', 26330),
 ('Samsung Galaxy S7 Edge', 23062)]

los articulos mas vistos estan de la forma (articulo, apariciones)